tests/rust/wasm32-wasip3/src/bin/filesystem-mkdir-rmdir.rs

```diff
@@ -0,0 +1,146 @@
+use std::process;
+extern crate wit_bindgen;
+
+wit_bindgen::generate!({
+    inline: r"
+  package test:test;
+
+  world test {
+      include wasi:filesystem/imports@0.3.0-rc-2025-09-16;
+      include wasi:cli/command@0.3.0-rc-2025-09-16;
+  }
+",
+    additional_derives: [PartialEq, Eq, Hash, Clone],
+    // Work around https://github.com/bytecodealliance/wasm-tools/issues/2285.
+    features:["clocks-timezone"],
+    generate_all
+});
+
+use wasi::filesystem::types::{Descriptor, DescriptorFlags, ErrorCode, OpenFlags, PathFlags};
+
+async fn test_mkdir_rmdir(dir: &Descriptor) {
+    let mkdir = |path: &str| dir.create_directory_at(path.to_string());
+    let rmdir = |path: &str| dir.remove_directory_at(path.to_string());
+
+    // create-directory-at: async func(path: string) -> result<_, error-code>;
+    assert_eq!(
+        dir.create_directory_at("".to_string()).await,
+        Err(ErrorCode::NoEntry)
+    );
+    assert_eq!(mkdir(".").await, Err(ErrorCode::Exist));
+    assert_eq!(mkdir("..").await, Err(ErrorCode::NotPermitted));
+    assert_eq!(mkdir("parent/foo").await, Err(ErrorCode::NotPermitted));
+    assert_eq!(mkdir("/").await, Err(ErrorCode::NotPermitted));
+    assert_eq!(
+        mkdir("../fs-tests.dir/q.cleanup").await,
+        Err(ErrorCode::NotPermitted)
+    );
+    assert_eq!(
+        mkdir("parent/fs-tests.dir/q.cleanup").await,
+        Err(ErrorCode::NotPermitted)
+    );
+    assert_eq!(mkdir("a.txt").await, Err(ErrorCode::Exist));
+    mkdir("q.cleanup").await.unwrap();
+    assert_eq!(
+        rmdir("../fs-tests.dir/q.cleanup").await,
+        Err(ErrorCode::NotPermitted)
+    );
+    assert_eq!(
+        rmdir("parent/fs-tests.dir/q.cleanup").await,
+        Err(ErrorCode::NotPermitted)
+    );
+    assert_eq!(
+        rmdir("q.cleanup/../../fs-tests.dir/q.cleanup").await,
+        Err(ErrorCode::NotPermitted)
+    );
+    rmdir("q.cleanup").await.unwrap();
+    mkdir("q.cleanup/").await.unwrap();
+    rmdir("q.cleanup").await.unwrap();
+    mkdir("q.cleanup").await.unwrap();
+    // FIXME: https://github.com/bytecodealliance/wasmtime/issues/11524
+    // rmdir("q.cleanup/")
+    //     .await.unwrap();
+    // mkdir("q.cleanup/////")
+    //     .await.unwrap();
+    // rmdir("q.cleanup////////////")
+    //     .await.unwrap();
+    // Using this instead to clean up:
+    rmdir("q.cleanup").await.unwrap();
+
+    // remove-directory-at: async func(path: string) -> result<_, error-code>;
+    assert_eq!(rmdir("").await, Err(ErrorCode::NoEntry));
+    assert_eq!(rmdir(".").await, Err(ErrorCode::Invalid));
```

Comment:
On Windows:
```
thread '<unnamed>' (1) panicked at src/bin/filesystem-mkdir-rmdir.rs:72:5:
assertion `left == right` failed
  left: Err(ErrorCode { code: 0, name: "access", message: "Permission denied, similar to `EACCES` in POSIX." })
 right: Err(ErrorCode { code: 11, name: "invalid", message: "Invalid argument, similar to `EINVAL` in POSIX." })
```

Reply:
Added an exception to allow `Access`.  Discouraging, but let's see if we manage to make it to the end of the file.